### PR TITLE
s4744924: Markus Zuegg Vision Transformer in Pytorch_lightning

### PR DESCRIPTION
**Pull request by MarkusZuegg**
Hello Teaching Team,

Here is my pull request to merge my topic-recognition into Shakes' topic-recognition. The task I picked was to create a Vision Transformer to classify Alzheimer's disease (AD) or Cognitive Normal (NC) in the dataset ADNI 2D MRI. I managed to get an accuracy of 64.27% in the test split. The files included in this PR are as followed:

readme_imgs/: folder contains images used in this README.md file

ViT_Pytorch_lightning/: folder contains all the following:

dataset.py: contains data loading class

modules.py: contains all neural network architecture

train.py: contains main ViT class and ability to train/save model and test

predict.py; contains scripts to load previous checkpoint and test that model
    
Data/: folder storing data should be located here

Thank you,
Markus Zuegg s4744924

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme:
Excellent

Commit messages: Excellent, detailed.

Code:
- Uses Vision transformer
- Well commented
- Good model design

Functionality/Performance:
- Test accuracy is 64.27%
- Did patient-level split
- Preprocssing data correctly
- Performed data augmentation
- Well done

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages sometimes -1
Progressive commits used
## Documentation
ReadMe acceptable
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
